test/tests/functional/pbs_offline_vnodes.py

```diff
@@ -86,39 +87,80 @@ def test_single_mom_hook_failure_affects_vnode(self):
         After this check if vnodes are marked offline.
         In case of a single mom reporting vnodes, it should mark
         all the vnodes and mom as offline.
+        Once offlined, reset the mom by issueing pbsnodes -r
+        and check if the job runs on one of the vnodes.
         """
-        self.create_multi_vnodes(1, True)
+        self.create_multi_vnodes(1)
         self.create_mom_hook()
+        # Remove the second node from the system (if any)
+        try:
+            self.server.manager(MGR_CMD_DELETE, NODE,
+                                id=self.moms.values()[1].shortname)
+        except PbsManagerError as e:
+            self.assertTrue("Unknown node" in e.msg[0])
 
         j1 = Job(TEST_USER)
         self.server.expect(NODE, {ATTR_NODE_state: 'free'},
-                           id=self.moms.values()[0].shortname)
+                           id=self.moms.values()[0].shortname, max_attempts=3,
+                           interval=2)
         j1.set_sleep_time(1000)
         jid = self.server.submit(j1)
-        self.server.manager(MGR_CMD_SET, SERVER, {'scheduling': 'False'})
         self.server.expect(JOB, {ATTR_state: 'Q'}, id=jid)
 
         self.server.expect(
             NODE, {ATTR_NODE_state: 'offline'},
-            id=self.moms.values()[0].shortname)
+            id=self.moms.values()[0].shortname, max_attempts=3, interval=2)
+
+        vname = "vnode[0]"
         if self.is_cray is True:
             vlist = []
+            first_vnode = 1
             vnl = self.server.filter(
                 VNODE, {'resources_available.vntype': 'cray_compute'})
             vlist = vnl["resources_available.vntype=cray_compute"]
             # Loop through each compute vnode in the list and check if state =
             # offline
             for v1 in vlist:
+                if first_vnode == 1:
+                    first_vnode += 1
+                    vname = v1
                 # Check that the vnode is in offline state
                 self.server.expect(
                     VNODE, {'state': 'offline'}, id=v1, max_attempts=3,
                     interval=2)
 
         else:
             self.server.expect(
-                NODE, {ATTR_NODE_state: 'offline'}, id='vnode[0]')
+                NODE, {ATTR_NODE_state: 'offline'}, id='vnode[0]',
+                max_attempts=3, interval=2)
             self.server.expect(
-                NODE, {ATTR_NODE_state: 'offline'}, id='vnode[1]')
+                NODE, {ATTR_NODE_state: 'offline'}, id='vnode[1]',
+                max_attempts=3, interval=2)
+
+        mom_host = self.moms.values()[0].shortname
+        pbs_exec = self.server.pbs_conf['PBS_EXEC']
+        pbsnodes_cmd = os.path.join(pbs_exec, 'bin', 'pbsnodes')
+        pbsnodes_reset = pbsnodes_cmd + ' -r ' + mom_host

```

Comment:
Shouldn't we also check that all of the nodes are free (i.e. no longer offline)?

Reply:
I think that is not needed because there wasn't a problem with "pbsnodes -r" not freeing up the nodes.

Reply:
Okay.  Thanks.